docs/site/migration/boot-scripts.md

```diff
@@ -6,8 +6,249 @@ sidebar: lb4_sidebar
+  // memory datasource in a sample LoopBack application created by CLI, regardless it's
+  // an LB3 app or LB4 app
+  const db = server.datasources.db;
+  console.log('This is a synchronized script.');
+  console.log('Your app has a datasource called: ', db.name);
+};
+```
+
+While a typical asynchronized script has a callback function as the second
+parameter:
+
+```js
+// a typical asynchronized boot script
+// in file `server/boot/create-sample.js`
+'use strict';
+
+module.exports = function createSample(server, done) {
+  // SUPPOSE your application has a Todo model
+  const Todo = server.models.Todo;
+  const sample = {title: 'a todo sample', desc: 'Something to do.'};
+  Todo.create(sample)
+    .then(result => {
+      console.log('Sample created as: ', result);
+    })
+    .then(done);
+};
+```
+
+Since LoopBack 4 application is created in TypeScript, which supports the latest
+ECMAScript features including
+[`async/await`](https://javascript.info/async-await), there is no such signature
+difference any more, the `start` function in the observer is always an `async`
+function.
+
+### Migrating to LoopBack 4 Observer
+
+To perform the same logic defined in a LoopBack 3 boot script, you should create
+a corresponding observer in your LoopBack 4 application.
+
+LoopBack 4 has its own life cycles at runtime and has `boot`, `start`, `stop` as
+its transition states. Observer allows you to participate in the `start` state,
+which is equivalent to the LoopBack 3 custom boot script.
+
+_Note: We highly recommend you learn about
+[Life cycle events and observers](https://loopback.io/doc/en/lb4/Life-cycle.html)
+first before reading the following guide._
+
+Take the two scripts(`server/boot/print-info.js` and
+`server/boot/create-sample.js`) above as examples, let's first migrate
+`server/boot/print-info.js` to your LoopBack 4 application.
+
+#### Creating Observer
+
+Run command `lb4 observer` to create an observer called `printInfo`:
+
+```sh
+lb4 observer
+? Observer name: printInfo
+? Observer group:
+   create src/observers/print-info.observer.ts
+   update src/observers/index.ts
+
+Observer PrintInfo was created in src/observers/
+```
+
+You will find a file called `PrintInfo.observer.ts` created in folder
+`src/observers/`. Open that file, there is a class called `PrintInfoObserver`,
+which implements `LifeCycleObserver` and has `start` and `stop` functions. The
+`start` function is where you should copy over the content of LB3's `printInfo`
+function.
+
+#### Injecting Application
+
+The LoopBack 3 boot script takes in `server` as its first parameter, which is
+essentially the `application` injected using key
+`CoreBindings.APPLICATION_INSTANCE`:
+
+```ts
+// ... imports
+export class PrintInfoObserver implements LifeCycleObserver {
+  constructor(
+    @inject(CoreBindings.APPLICATION_INSTANCE) private app: Application,
+  ) {
+    // ... other class members
+  }
+```
```

Comment:
Please add `}` to close the class definition.

The markdown & typescript parser & highlighter used by GitHub is confused now and shows the text below in blue-ish color, while regular text should be rendered in black (see the text before this code snippet).

Reply:
oops